### PR TITLE
Replace AppVeyor and Travis-CI with GitHub Actions.

### PR DESCRIPTION
**Pull request by dblock**
Replaces Travis and AppVeyor CI by GitHub Actions where we can run CI for all platforms. AppVeyor and Travis can/should be turned off before merge in repo settings (I think there are permissions for those that should be removed).

Passing build: https://github.com/dblock/JenkinsPipelineUnit/actions/runs/1449798899

Closes https://github.com/jenkinsci/JenkinsPipelineUnit/issues/424

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Comment by nre-ableton** [comment]
@dblock Do we need to enable something for GitHub Actions in order to see them working on this PR?

**Comment by dblock** [comment]
> @dblock Do we need to enable something for GitHub Actions in order to see them working on this PR?

I think you would have to merge this, but you would see the same thing as https://github.com/dblock/JenkinsPipelineUnit/actions/runs/1449798899

AppVeyor would have to be disabled in repo config to avoid it showing a failure, too.

**Review by nre-ableton — COMMENTED** [review]
@dblock Can you please rebase this PR and resolve conflicts? Sorry for the delay, but let's go ahead and merge this.

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Setup Java
+        uses: actions/setup-java@v1
+        with:
+          java-version: 8
+      - uses: actions/checkout@v2
+      - name: Run Tests
+        run: |
+          ./gradlew test --info
```
Comment (nre-ableton) [inline .github/workflows/tests.yml:22]: Nitpick: if you could add a newline here, that would be nice. :smile:

**Comment by nre-ableton** [comment]
Actually @dblock, let's put this PR on hold until https://github.com/jenkins-infra/helpdesk/issues/2776 is resolved. Sorry for the noise!

**Comment by dblock** [comment]
> Actually @dblock, let's put this PR on hold until [jenkins-infra/helpdesk#2776](https://github.com/jenkins-infra/helpdesk/issues/2776) is resolved. Sorry for the noise!

No problem. LMK if/when you want this. Honestly IMO GHA is *so* much easier to use. I see no problem using GHA for building other build tools.

**Comment by nre-ableton** [comment]
@dblock Thanks for the reminder. Fixed in f020a30b6dfe268ece03ec15e7c46a97337d1800.

**Pull request closed without merging**